nml/actions/action2layout.py

```diff
@@ -216,7 +216,7 @@ def get_sprite_number(self):
         if add_sprite:
             expr = expression.BinOp(nmlop.ADD, sprite, expr, sprite.pos)
         if add_palette:
-            expr = expression.BinOp(nmlop.ADD, palette, expr, sprite.pos)
+            expr = expression.BinOp(nmlop.ADD, expression.BinOp(nmlop.SHIFT_LEFT, palette, expression.ConstantNumeric(16, sprite.pos), sprite.pos), expr, sprite.pos)
         return expr.reduce()
```

Comment:
The conditionals and constant arithmetic about `add_sprite` and `add_palette` can be removed; `BinOp.reduce()` does this anyway.